packages/react/src/modules/core/components/Switch/Switch.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { useState } from "react";
+import "./styles.css";
+
+interface SwitchProps {
+  isChecked?: boolean;
+  onColor?: string;
+  onChange?: () => void;
+}
+
+export const Switch: React.FC<SwitchProps> = ({
+  isChecked = false,
+  onColor,
+  onChange,
+}) => {
+  console.log("is Checked", isChecked)
+  const [on, setOn] = useState(isChecked || false);
+
+  const toggleSwitch = () => {
+    if (onChange) {
+      onChange();
+      setOn(!on);
```

Comment:
Nit: could simplify logic to only call `setOn` once (outside of the if statement)